packages/components/src/color-picker/use-deprecated-props.ts

```diff
@@ -109,39 +107,37 @@ const transformColorStringToLegacyColor = memoize(
 export function useDeprecatedProps(
 	props: LegacyProps | ColorPickerProps
 ): ColorPickerProps {
+	const isUsingLegacy = isLegacyProps( props );
+	const { onChangeComplete } = props as LegacyProps;
+	const { onChange: onChangeProp } = props as ColorPickerProps;
 	const onChange = useCallback(
 		( color: string ) => {
-			if ( isLegacyProps( props ) ) {
-				return props.onChangeComplete(
+			if ( isUsingLegacy ) {
+				return onChangeComplete(
 					transformColorStringToLegacyColor( color )
 				);
 			}
 
-			return props.onChange?.( color );
+			return onChangeProp?.( color );
 		},
-		[
-			( props as LegacyProps ).onChangeComplete,
-			( props as ColorPickerProps ).onChange,
-		]
+		[ onChangeComplete, onChangeProp, isUsingLegacy ]
 	);
 
+	const { color: colorProp } = props;
 	const color = useMemo( () => {
-		return isLegacyProps( props )
-			? getColorFromLegacyProps( props )
-			: props.color;
-	}, [ props.color ] );
+		return isUsingLegacy
+			? getColorFromLegacyProps( colorProp )
+			: ( colorProp as ColorPickerProps[ 'color' ] );
+	}, [ colorProp, isUsingLegacy ] );
 
+	const { disableAlpha } = props as LegacyProps;
+	const { enableAlpha: enableAlphaProp } = props as ColorPickerProps;
 	const enableAlpha = useMemo( () => {
-		return isLegacyProps( props )
-			? ! props.disableAlpha
-			: props.enableAlpha;
-	}, [
-		( props as LegacyProps ).disableAlpha,
-		( props as ColorPickerProps ).enableAlpha,
-	] );
+		return isUsingLegacy ? ! disableAlpha : enableAlphaProp;
+	}, [ disableAlpha, enableAlphaProp, isUsingLegacy ] );
```

Comment:
I'd say that both these `useMemo`s are likely adding more overhead than they're saving.

Reply:
Makes sense to me. f0873ec removes them. Hope that was the actual suggestion and not just musing 😁.